### PR TITLE
Properly handle the autocompletions list when switching tabs

### PR DESCRIPTION
**Pull request by victorporof**
Fixes https://github.com/mozilla/tofino/issues/1087

Depends on https://github.com/mozilla/tofino/pull/1079

Signed-off-by: Victor Porof <vporof@mozilla.com>

This dependency graph might help: 
screen shot 2016-09-12 at 12 40 55 pm

### CODE REVIEW

File: app/ui/shared/widgets/autocompleted-search.jsx [hunk 1]
```diff
@@ -69,9 +70,9 @@ class AutocompletedSearch extends Component {
         }
         break;
       case 'ArrowUp':
-        if (showAutocompletions && autocompletionResults.size) {
+        if (showAutocompletions && hasAutocompletionResults) {
           if (selectedIndex === 0) {
-            this.setState({ selectedIndex: autocompletionResults.size - 1 });
+            this.setState({ selectedIndex: hasAutocompletionResults - 1 });
```
Comment (bgrins) [inline app/ui/shared/widgets/autocompleted-search.jsx]: This is meant to be an int

File: app/ui/shared/widgets/autocompleted-search.jsx [hunk 2]
```diff
@@ -81,8 +82,8 @@ class AutocompletedSearch extends Component {
         break;
       case 'Tab':
       case 'ArrowDown':
-        if (showAutocompletions && autocompletionResults.size) {
-          if (selectedIndex === autocompletionResults.size - 1) {
+        if (showAutocompletions && hasAutocompletionResults) {
+          if (selectedIndex === hasAutocompletionResults - 1) {
```
Comment (bgrins) [inline app/ui/shared/widgets/autocompleted-search.jsx]: Ditto

File: app/ui/shared/widgets/autocompleted-search.jsx (hunk 1 above)
Comment (victorporof) [inline app/ui/shared/widgets/autocompleted-search.jsx]: Oops!

Comment (victorporof) [inline app/ui/shared/widgets/autocompleted-search.jsx]: Well, it is an int, but the variable is poorly named.

**Comment by bgrins** [comment]
The fact that we keep it open with the old results is interesting and different from other browsers.  In particular, afaict other browser treat the autocomplete list *not* as per-page but a single global thing that gets closed and cleared when switching tabs.

But I think it leads to some UX things, like when I do:

* Open two tabs
* Type something in URL bar A to show autocomplete list
* Switch tabs (ctrl+tab)
* Type something in URL bar B to show autocomplete list
* Switch back (ctrl+tab)

The autocomplete isn't focused but it looks like it is (i.e. pressing down arrow doesn't switch between results).

Another existing issue that I'll file separately: the list doesn't disappear even when I click elsewhere in the browser (on another tab or even just somewhere in the nav bar).  Seems the only way to get it to go away is by pressing ESC - but once I've clicked elsewhere the URL bar loses focus so ESC doesn't close it.

**Comment by bgrins** [comment]
I'm OK to handle the focus issue in https://github.com/mozilla/tofino/pull/1091#issuecomment-247046366 as a follow up if you want to track that separately

**Comment by victorporof** [comment]
@bgrins the focus issue was there even before this patch. Indeed it would be a great idea to support it to make the autocompletion-list-per-page thing nice from a UX pov.

**Pull request merged**